src/server/api/endpointRouter/endpointRouterController.js

```diff
@@ -9,59 +9,80 @@ var getData = function(req, res, next) {
   var route = req.params[0];
   var method = req.method;
   var data;
+  
+  var date = new Date();
+  var day = date.getDay();
+  if(day === 0) {
+    day = 7;
+  }
 
   Endpoint.findOne({ 'username': username, 'route': route }, function (err, endpoint) {
     if (err) return res.status(500).end(err);
     if (!endpoint) {
       return res.status(500).end(err);
     } else {
-      //if persistance is set to true, we let the user persist data through their API endpoint
-      if(endpoint.persistence === true) {
-
-        //if we have params, get specific data from data inserted through API created
-        if (req.query.id) {
-          var queryID = parseInt(req.query.id);
-          for (var i=0; i<endpoint.data.length; i++) {
-            var dataPoint = endpoint.data[i];
-            if (dataPoint.id === queryID) {
-              return res.status(200).json(dataPoint);
+      if (!endpoint.analytics) {
+        endpoint.analytics = {};
+      }
+      if(!endpoint.analytics[method]) {
+        endpoint.analytics[method] = {};
+      }
```

Comment:
would it make more sense to always initialize a new endpoint with an `analytics` property with all the methods set on it?